sklearn/utils/validation.py

```diff
@@ -690,6 +705,14 @@ def check_X_y(X, y, accept_sparse=False, accept_large_sparse=True,
     y_converted : object
         The converted and validated y.
     """
+    # warn_on_dtype deprecation
```

Comment:
I think that you don't need to warn here since the warning occurs in the call to `check_array()`.

That would warn twice.

Reply:
I agree, this is removed from check_X_y